src/java.base/share/classes/jdk/internal/foreign/MemorySessionImpl.java

```diff
@@ -57,9 +58,9 @@ public abstract sealed class MemorySessionImpl
         permits ConfinedSession, GlobalSession, SharedSession {
     static final int OPEN = 0;
     static final int CLOSED = -1;
+    static final int NONCLOSEABLE = 1;
```

Comment:
Nice trick! This effectively relegates 0 (open) to be the default value of the stable state field. That will either go to -1 (closed) or 1 (non-closeable) and be treated as constant thereafter. Perhaps worth adding a comment on the rationale behind this.